website/code_issues.qmd

```diff
@@ -128,8 +128,91 @@ foo <- function(..., verbose = TRUE){
 
 Functions can print per default, like the example above, as long as the printing can be turned off (here, by setting `verbose = FALSE`).
 
+:::{.callout-note}
+
+`print()` and `cat()` are not the only functions which can write output onto the console. The issue described in the recipe, also applies to the use of other printing function like `writeLines()`. 
+If you are using loggers to document your functions' process, make sure that users can set their log level such that not messages are displayed.
+
+:::
+
 
 
 ---------------------------------------------------------
 
-# Change of options() and par() 
\ No newline at end of file
+# Change of Options, Graphical Parameters and Working Directory
+
+## Problem
+
+You are changing the `par()`, `options()` or `setwd()` in your functions, examples, demos or vignettes without resetting them.
+
+## Solution 
+
+Reset the changed options, in your functions by using [`on.exit()`](https://stat.ethz.ch/R-manual/R-devel/library/base/html/on.exit.html) or in examples, demos and vignettes with an additional line of code after the example.
+
+### Details
+
+Ideally, the user's options are not changed at all. If they really have to be altered, restoring the previous values of user options is mandatory for CRAN packages. There are different ways of resetting for functions, and examples, demos or vignettes which are recommended by CRAN. 
+
+Changing `par()`, `options()` or `setwd()` all invisibly return the previous values and therefore these can be stored in variables using the assignment operator `<-` and later be restored by calling the variable name as argument in the respective function. 
+
+
+**For functions**:
+
+When changing options inside one of your package functions, you can use [`on.exit()`](https://stat.ethz.ch/R-manual/R-devel/library/base/html/on.exit.html) for restoring. 
```

Comment:
Perhaps mention what to do when multiple settings are set and need to be removed by `on.exit()` that then one needs to use `on.exit(add = TRUE)`. Or at least encourage to read the documentation of the function.

Reply:
Good point. 
I already have the `.callout-tip` at the end of this recipe:

> If you need to change more than one option in the same function, example, vignette or demo, you can use `oldpar <- par(no.readonly = TRUE)` or `oldop <- options()` to reset all parameters at once. Saving the entire 
> Note, that for `par()` the `no.readonly` argument must be set to `TRUE`, else warnings will be produced.

I will move up a bit and add a sentence to encourage reading the documentaions.